tests/unittests/Foundation/NSArrayTests.mm

```diff
@@ -425,3 +425,13 @@ static unsigned long int objectIndexInArray(NSArray* array, int value, int start
 
     EXPECT_OBJCNE(input, output);
 }
+
+TEST(NSMutableArray, InsertingNilShouldThrow) {
+    NSMutableArray* arr = [NSMutableArray arrayWithObject:@"hello"];
+    EXPECT_ANY_THROW(arr[0] = nil);
+    EXPECT_ANY_THROW([arr insertObject:nil atIndex:1]);
+    EXPECT_ANY_THROW([arr addObject:nil]);
```

Comment:
It absolutely is.
The only useful information we would get out of this is that it is a *NSInvalidArgumentException*. Unfortunately, that's not fine-grained enough to support what we care about.

"Any" exception is fine.